[jit] kill `shallowEquals`

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #35747 [jit] make node hashing use ivalue equality
* #35019 [jit] kill isSameIdentity
* **#35005 [jit] kill `shallowEquals`**
* #34986 [jit] define equality for IValue

This is one of the ad-hoc IValue equality implementations that should be
replaced with `operator==`.

Differential Revision: [D20537900](https://our.internmc.facebook.com/intern/diff/D20537900)

suo committed:
[jit] kill `shallowEquals`

This is one of the ad-hoc IValue equality implementations that should be
replaced with `operator==`.

[ghstack-poisoned]

suo committed:
Update on "[jit] kill `shallowEquals`"


This is one of the ad-hoc IValue equality implementations that should be
replaced with `operator==`.

[ghstack-poisoned]

suo committed:
Update on "[jit] kill `shallowEquals`"


This is one of the ad-hoc IValue equality implementations that should be
replaced with `operator==`.

[ghstack-poisoned]

suo committed:
Update on "[jit] kill `shallowEquals`"


This is one of the ad-hoc IValue equality implementations that should be
replaced with `operator==`.

[ghstack-poisoned]

zdevito commented on torch/csrc/jit/mobile/register_mobile_ops.cpp:
```diff
@@ -94,13 +94,13 @@ void cat_kernel(const c10::OperatorHandle& op, Stack* stack) {
   c10::IValue self, obj;
   pop(*stack, self, obj);
-  push(*stack, self.is(obj));
+  push(*stack, self.isSameIdentity(obj));
```
Why is this no longer `is`?

suo committed:
Update on "[jit] kill `shallowEquals`"


This is one of the ad-hoc IValue equality implementations that should be
replaced with `operator==`.

Differential Revision: [D20537900](https://our.internmc.facebook.com/intern/diff/D20537900)

[ghstack-poisoned]

suo committed:
Update on "[jit] kill `shallowEquals`"


This is one of the ad-hoc IValue equality implementations that should be
replaced with `operator==`.

Differential Revision: [D20537900](https://our.internmc.facebook.com/intern/diff/D20537900)

[ghstack-poisoned]

suo committed:
Update on "[jit] kill `shallowEquals`"


This is one of the ad-hoc IValue equality implementations that should be
replaced with `operator==`.

Differential Revision: [D20537900](https://our.internmc.facebook.com/intern/diff/D20537900)

[ghstack-poisoned]
